src/Microsoft.OData.Client/Materialization/MaterializerEntry.cs

```diff
@@ -263,7 +263,7 @@ public static MaterializerEntry GetEntry(ODataResource entry)
         /// <param name="link">The link.</param>
         public void AddNestedResourceInfo(ODataNestedResourceInfo link)
         {
-            if (this.IsTracking)
+            if (this.IsTracking && !this.Entry.IsTransient)

```

Comment:
@KenitoInc @habbes A typical aggregation response looks as follows:
```json
{
    "@odata.context": "http://tempuri.org/$metadata#Sales(Customer(Country),AverageAmount)",
    "value": [{
        "odata.id": null,
        "AverageAmount": 3.500000,
        "Customer": {"odata.id":null,"Country":"Netherlands"}
    }]
}
```
The `odata.id` annotation has a value of `null`. What happens in ODL is that when such a resource is being serialized, the `IsTransient` property of the `ODataResource` is set to `true`.
Even when tracking is enabled, a transient entity resource [should NOT be added to the entity descriptor](https://github.com/OData/odata.net/blob/4e263ca4b94d01f2d105d8282c62ec272cfcd1bb/src/Microsoft.OData.Client/Materialization/FeedAndEntryMaterializerAdapter.cs#L362-L364). That's what I fixed here since it was causing an error. Without that fix, the logic would try to track `Country` as a nested resource and fail catastrophically since there isn't enough information - specifically `link.Url` and `link.AssociationLinkUrl` are uninitialized.
We were already doing a [similar check here](https://github.com/OData/odata.net/blob/4e263ca4b94d01f2d105d8282c62ec272cfcd1bb/src/Microsoft.OData.Client/Materialization/FeedAndEntryMaterializerAdapter.cs#L362-L364) for the resource itself. I believe failing to do it for the nested resource was an oversight